### PR TITLE
Add support for pro pricing

### LINKED ISSUE #134: Pro product features for collections pages
Collections pages need to denote when a product is "Pro Only" as well as show the Pro price to users logged in with a Pro discountid.

https://www.ifixit.com/Store/iPhone/iPhone-6/Screens
Screen Shot 2021-11-18 at 1 04 29 PM

Pros are denoted by a `discountid` of 13, 14, 15, 17 or 18 in `ifixit_cart.customers`

`reseller_pricing_enabled` (attributeid 49) in `ifixit_cart_attributes` denotes if Pros get a tiered price or a standard %

- If `reseller_pricing_enabled=1` then customers get the corresponding tiered price. `attributeid` 50, 51, 52 and 53 are the prices for each tier. 
- If `reseller_pricing_enabled=0` then customers get a standard % off according to `ifixit_cart.customer_discount`

### PR DESCRIPTION
**Pull request by dhmacs**
fixes #134 

## Changelog

- Show pro pricing to pro users
- Hide the price range filter when a user is pro
- Show pro badge for pro products

## QA

1. Visit the Vercel PR preview (replace `vercel.app` with `cominor.com`)
2. Visit the tools product list (`/Store/Tools`)
3. Log in as a pro user in cominor (e.g. `pro_1@ifixit.com` with pwd `iamapro`)
4. Verify that products marked as PRO have a PRO badge (e.g. `Surface Display Debonding Tool`)
5. Verify that pro prices are displayed
6. Verify that the price range filter is not available

### CODE REVIEW

**Comment by jordycosta** [comment]
⚠️ It seems that the `PRO` tag is displayed for all user levels

> I see no price range or missing `PRO` tag when I am signed out or signed into an account without a Pro tag attribute

result

**Comment by dhmacs** [comment]
@jordycosta 

> ⚠️ It seems that the `PRO` tag is displayed for all user levels

The live site (https://www.ifixit.com/Store/Tools/iFixit_Exclusives?page=2&display=list) shows PRO badges even if the user is logged out. Should we change this?

Screenshot 2022-04-14 at 08 41 18

**Comment by jordycosta** [comment]
QA 🟢 

Show pro pricing to `Pro` users ☑️ 
Price range filter not displayed for `Pro` users ☑️ 
`Pro` badge shown for Pro products ☑️

---

### Testing Procedure:
1. Visit the Vercel [preview](https://react-commerce-git-add-pro-pricing-support-ifixit.cominor.com/) 
2. Sign in as a pro user
3. Ensure the price range filter is no longer shown
4. Search for a product that has pro pricing (e.g. `Mako Driver Kit - Refurbished`)
5. Check that the price is discounted in comparison to non-Pro users
6. Ensure that the Pro tag is displayed for Pro-only products (e.g. `Surface Display Debonding Tool`)
 
---

### Screenshots:

signed_in_pro

signed_in_non_pro

result

**Comment by jordycosta** [comment]
> @jordycosta
> 
> > ⚠️ It seems that the `PRO` tag is displayed for all user levels
> 
> The live site (https://www.ifixit.com/Store/Tools/iFixit_Exclusives?page=2&display=list) shows PRO badges even if the user is logged out. Should we change this?
> 
> Screenshot 2022-04-14 at 08 41 18

I believe that it's intended to be shown 👍

**Comment by erinemay** [comment]
deploy_block 🦑 just to clarify... @jordycosta, the debonding tool is only showing up for logged in pro users, right? Or are you seeing it for non-pro users? I don't see that in our collection pages or search results on ifixit.com. I think because it's a pro-only part without any non-pro variants?
image

**Review by davidrans — APPROVED** [review]
CR :coffee:

**Comment by dhmacs** [comment]
@erinemay 

> deploy_block 🦑 just to clarify... @jordycosta, the debonding tool is only showing up for logged in pro users, right? Or are you seeing it for non-pro users? 

I can answer this, it's being show to everyone. When you click it leads to this page https://www.cominor.com/Store/Tools/Surface-Display-Debonding-Tool-M1214770-001/IF265-010?o=1

If you want this to be hidden, what's the criteria that I should use in Algolia to achieve this?

**Comment by erinemay** [comment]
> If you want this to be hidden, what's the criteria that I should use in Algolia to achieve this?

Great question. I'm not sure. @sterlinghirsh?

File: frontend/lib/strapi-sdk/generated/sdk.ts [hunk 1]
```diff
@@ -1,1357 +1,2347 @@
 export type Maybe<T> = T | null;
-export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
-export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
-export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
+export type Exact<T extends { [key: string]: unknown }> = {
```
Comment (davidrans) [inline frontend/lib/strapi-sdk/generated/sdk.ts:2]: What's up with this strapi diff?

Comment (dhmacs) [reply frontend/lib/strapi-sdk/generated/sdk.ts:2]: This is an auto generated file... Changes are due to formatting. It's weird that it was different before, but it should be correct now

**Comment by lithobraking** [comment]
> If you want this to be hidden, what's the criteria that I should use in Algolia to achieve this?

I don't think we have a flag like that currently; we'll probably have to add one.

**Review by davidrans — APPROVED** [review]
CR :coffee:

**Comment by sterlinghirsh** [comment]
OK I looked again and the is_pro field indicates whether a product is pro only. When a user is logged out or logged in as a non-pro, add a filter `is_pro = 0` similar to how we do public = 1. When a user is logged in as a pro, omit that filter so that any value is allowed for is_pro. Pros will see both pro-only and normal products, but the pro-only ones get the PRO flair. The flair looks right in this pull, so the only necessary change is adding the filter for is_pro = 0.

**Comment by sterlinghirsh** [comment]
CR :zap: We can ship it as-is and add the pro filter in another pull, or we can include the pro filter in this pull. Up to you.

**Comment by dhmacs** [comment]
@sterlinghirsh Let's merge this and do that fix in another PR

**Comment by jordycosta** [comment]
QA 🟢 

Show pro pricing to `Pro` users ☑️ 
Price range filter not displayed for `Pro` users ☑️ 
`Pro` badge shown for Pro products ☑️

**Comment by dhmacs** [comment]
@erinemay are you ok with merging this?

**Comment by davidrans** [comment]
I think it's fine. If we want to change up the spec for how we label pro products, we can fix it in a follow-up pull

**Pull request merged**